### DISCUSSION TITLE
Generate Output base line document

### DISCUSSION THREAD

**Discussion by kiko0217** [mindee/doctr#908] [General]
Hai, Thank you for useful package. 
of most requirements, many projects require ocr to read documents in paragraph or line order. maybe can be added?

**Comment by felixdittrich92** [comment]
Hi @kiko0217 👋 ,
can you please explain a bit more what you expect ? :)
So the docTR output provides currently a block, line, word structure

**Reply by info2000** [reply]
@kiko0217 actually you can build the final text with this code

`json_output = result.export()
pagetext=""
for key in json_output['pages']:
  print('page', key['page_idx'])
  for bloque in key['blocks']:
    for line in bloque['lines']:
      for word in linea['words']:
        pagetext+=word['value']+" "
      pagetext+=". "
    pagetext+="\n\n"

print(pagetext)`

and will have a readable output text

**Reply by kiko0217** [reply]
> Hi @kiko0217 👋 , can you please explain a bit more what you expect ? :) So the docTR output provides currently a block, line, word structure

I mean, maybe because the current output structure document, page, block, line, word. but in actual usage needed structure document page, and then line. so must create code to generate output as needed.
example code.
```
import os
os.environ['USE_TORCH']= '1'
from doctr.io import DocumentFile, Line
from doctr.models import ocr_predictor
from glob import glob
import time

def main(args):
    t0= time.time()
    dir_images = glob(args.images)
    dir_images = list(filter(lambda x: (x.lower().endswith(('.png', '.jpg', '.jpeg', '.tiff', '.bmp', '.gif'))), dir_images))
    doc = DocumentFile.from_images(dir_images)
    model = ocr_predictor(pretrained=True)
    result = model(doc)
    # print(result.render(page_break='\n\nresult:\n'))
    for page in result.pages:
        lines = []
        for block in page.blocks:
            for line in block.lines:
                lines.append(line)
        
        lines = sorted(lines, key=lambda x: [x.geometry[0][1], x.geometry[0][0]])
        
        new_lines=[]
        index_lines = 0
        for i, line in enumerate(lines):
            if (i == 0):
                new_lines.append(line)
                continue
            if(line.geometry[0][1]-new_lines[index_lines].geometry[0][1] > 0.012):
                new_lines.append(line)
                index_lines+=1
                continue
            new_words = new_lines[index_lines].words+line.words
            new_words = sorted(new_words, key=lambda x: x.geometry[0][0])
            new_lines[index_lines] = Line(words=new_words)
        for line in new_lines:
            print(line.render())
        print('\n\n\n')
    t0=time.time()-t0
    print(f'time; {t0} s')

def parse_args():
    import argparse
    parser = argparse.ArgumentParser(description='DocTR training script for text recognition (PyTorch)',
                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
                                     
    parser.add_argument('images', type=str, default=None, help='path images')
    
    args = parser.parse_args()

    return args

if __name__ == "__main__":
    args = parse_args()
    main(args)
```

so if docTr it can generate in output, maybe i think it's userful.

**Reply by frgfm** [reply]
HI @kiko0217 👋 

I don't get the point of your last reply? If someone has specific layout constraints, it's easier for that person to use a finer hierarchy to a looser one than the opposite :) 

Here we have document/page/blocks/lines/words, so you can easily extract any subset of those

If we do document/page/lines/words, then people interested in paragraph (=blocks), they need to do much more than filter outputs. So back to the beginning of this issue, could you specify what you suggestion is please? 🙏
